crates/data_privacy/src/data_class.rs

```diff
@@ -60,3 +75,228 @@ impl IntoDataClass for DataClass {
         self
     }
 }
+
+#[cfg(feature = "serde")]
+impl Serialize for DataClass {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        serializer.collect_str(self)
+    }
+}
+
+#[cfg(feature = "serde")]
+impl<'de> Deserialize<'de> for DataClass {
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        struct DataClassVisitor;
+
+        impl de::Visitor<'_> for DataClassVisitor {
+            type Value = DataClass;
+
+            fn expecting(&self, formatter: &mut core::fmt::Formatter) -> core::fmt::Result {
+                formatter.write_str("a string in taxonomy/name format")
+            }
+
+            fn visit_str<E: de::Error>(self, v: &str) -> Result<Self::Value, E> {
+                let (taxonomy, name) = v
+                    .split_once('/')
+                    .ok_or_else(|| de::Error::custom("expecting taxonomy/name format"))?;
+
+                if !is_valid_identifier(taxonomy) {
+                    return Err(de::Error::custom("invalid taxonomy identifier"));
+                }
+
+                if !is_valid_identifier(name) {
+                    return Err(de::Error::custom("invalid name identifier"));
+                }
+
+                Ok(DataClass {
+                    taxonomy: Cow::Owned(taxonomy.to_owned()),
+                    name: Cow::Owned(name.to_owned()),
+                })
+            }
+        }
+
+        deserializer.deserialize_str(DataClassVisitor)
+    }
+}
+
+/// Checks if a byte is a valid ASCII start character for a Rust identifier.
+const fn is_valid_ascii_ident_start(byte: u8) -> bool {
+    matches!(byte, b'a'..=b'z' | b'A'..=b'Z' | b'_')
+}
+
+/// Checks if a byte is a valid ASCII continuation character for a Rust identifier.
+const fn is_valid_ascii_ident_continue(byte: u8) -> bool {
+    matches!(byte, b'a'..=b'z' | b'A'..=b'Z' | b'0'..=b'9' | b'_')
+}
+
+/// Validates that a string is a valid Rust identifier (ASCII only).
+///
+/// This supports standard ASCII identifiers: `foo`, `_bar`, `Baz123`
+///
+/// Valid identifiers must:
+/// - Start with `_` or an ASCII letter (a-z, A-Z)
```

Comment:
why do we allow starting with `_`?

Reply:
I wanted Rust-compatible identifiers and _ is valid Rust. I opted not to support the full Unicode set of allowed identifier characters however, since that pulls in a ton of Unicode-related crates. So I support the ASCII subset.